diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9d516d5..3502a27 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,17 +71,12 @@ IF(NOT EXISTS ${FPARSER_ROOT_DIR})
     SET(FPARSER_ROOT_DIR /usr)
 ENDIF()
 message(STATUS "FPARSER_ROOT_DIR: ${FPARSER_ROOT_DIR}" )
-find_library(fparser_LIBRARIES 
-  NAMES fparser
-  HINTS ${FPARSER_ROOT_DIR}/lib${LIB_SUFFIX}
-  NO_CMAKE_FIND_ROOT_PATH
-)
-message(STATUS "fparser_LIBRARIES: ${fparser_LIBRARIES}" )
-find_path(fparser_INCLUDE_DIR 
+find_path(fparser_INCLUDE_DIR
   NAMES fparser.hh
-  HINTS ${FPARSER_ROOT_DIR}/include ${FPARSER_ROOT_DIR}/include/fparser
+  HINTS ${FPARSER_ROOT_DIR}
   NO_CMAKE_FIND_ROOT_PATH
 )
+set(FPARSER_SOURCES ${FPARSER_ROOT_DIR}/fparser.cc ${FPARSER_ROOT_DIR}/fpoptimizer.cc)
 message(STATUS "fparser_INCLUDE_DIR: ${fparser_INCLUDE_DIR}" )
 #TODO test if fparser was found
 INCLUDE_DIRECTORIES( ${fparser_INCLUDE_DIR} ) 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 8903a60..996f85c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -83,10 +83,9 @@ set(SOURCES
 )
 
 # CSXCAD library
-add_library( CSXCAD SHARED ${SOURCES} )
+add_library( CSXCAD SHARED ${SOURCES} ${FPARSER_SOURCES})
 
 TARGET_LINK_LIBRARIES( CSXCAD
-  ${fparser_LIBRARIES}
   ${TinyXML_LIBRARIES}
   ${HDF5_LIBRARIES}
   ${HDF5_HL_LIBRARIES}
